modules/samples/oauth2/playground2/src/main/org/wso2/sample/identity/oauth2/IDTokenDecrypterServlet.java

```diff
@@ -0,0 +1,129 @@
+        } else if (StringUtils.isBlank(idToken)) {
+            response.setStatus(HttpStatus.SC_BAD_REQUEST);
+            out.print("Error occurred while decrypting: Empty id token received!");
+        } else {
+            response.setContentType("application/json");
+
+            EncryptedJWT encryptedJWT;
+            try {
+                encryptedJWT = decryptJWE(idToken, privateKeyString);
+                JSONObject outJSON = new JSONObject();
+                JSONObject claimsJSON = new JSONObject();
+
+                // Get all claims set to a map and return a JSON object.
+                Map<String, Object> allClaims = encryptedJWT.getJWTClaimsSet().getAllClaims();
+                for (Map.Entry<String, Object> entry : allClaims.entrySet()) {
+                    claimsJSON.put(entry.getKey(), entry.getValue());
+                }
+                outJSON.put("claims", claimsJSON);
+
+                // Get JWT header data.
+                outJSON.put("header", encryptedJWT.getHeader().toJSONObject());
+
+                out.print(outJSON.toString());
+            } catch (NoSuchAlgorithmException | ParseException | JOSEException | IllegalArgumentException e) {
+                LOGGER.log(Level.SEVERE, e.getMessage(), e);
+                response.setStatus(HttpStatus.SC_BAD_REQUEST);
+                out.print("Error occurred while decrypting id token.");
+            } catch (InvalidKeySpecException e) {
+                LOGGER.log(Level.SEVERE, e.getMessage(), e);
+                response.setStatus(HttpStatus.SC_BAD_REQUEST);
+                out.print("Invalid client private key.");
+            }
+        }
+    }
+
+    /**
+     * Decrypt the id token using the private key.
+     *
+     * @param JWE              id token to be decrypted
+     * @param privateKeyString client private key as a string
+     * @return decrypted id token as an EncryptedJWT object
+     * @throws NoSuchAlgorithmException
+     * @throws InvalidKeySpecException
+     * @throws ParseException
+     * @throws JOSEException
+     * @throws IllegalArgumentException
+     */
+    private EncryptedJWT decryptJWE(String JWE, String privateKeyString)
+            throws NoSuchAlgorithmException, InvalidKeySpecException, ParseException, JOSEException, IllegalArgumentException {
+
+        KeyFactory kf = KeyFactory.getInstance("RSA");
+        // Remove EOF characters from key string and generate key object.
+        privateKeyString = privateKeyString.replace("\n", "").replace("\r", "");
+        PKCS8EncodedKeySpec keySpecPKCS8 = new PKCS8EncodedKeySpec(Base64.getDecoder().decode(privateKeyString));
+        PrivateKey privateKey = kf.generatePrivate(keySpecPKCS8);
+
+        EncryptedJWT jwt = EncryptedJWT.parse(JWE);
+
+        // Create a decrypter with the specified private RSA key.
+        RSADecrypter decrypter = new RSADecrypter((RSAPrivateKey) privateKey);
+
+        jwt.decrypt(decrypter);
+
+        return jwt;
```

Comment:
In this method, actually what is happening is returning a decrypted jwt.
But the return type is set to 'EncryptedJWT'. What is actually happening(decrypting) and the retrun value(Encrypted) seems contradictory.
Can you please explain this behavior? or can we fix it properly to match the flow.